src/flp/gadgets.rs

```diff
@@ -35,7 +35,7 @@ impl<F: FieldElement> Mul<F> {
     /// called by the validity circuit.
     pub fn new(num_calls: usize) -> Self {
         let n = gadget_poly_fft_mem_len(2, num_calls);
-        let n_inv = F::from(F::Integer::try_from(n).unwrap()).inv();
+        let n_inv = F::try_from(F::Integer::try_from(n).unwrap()).unwrap().inv();
```

Comment:
Propagating errors is more awkward here since these `::new()` functions are infallible. Could we at least include a comment explaining why this is expected to be infallible?